main/src/library/Iterator.flix

```diff
@@ -20,6 +20,60 @@ pub enum Iterator[a] {
 
 namespace Iterator {
 
+    ///
+    /// Returns `Some(x)` if `x` is the first element of `iter`.
+    ///
+    /// Returns `None` if `iter` is empty.
+    ///
+    /// Consumes the first element of the iterator.
+    ///
+    pub def head(iter: Iterator[a]): Option[a] & Impure = ???
+
+    ///
+    /// Returns an iterator of all integers between `b` (inclusive) and `e` (exclusive).
+    ///
+    /// Returns `Done` if `b >= e`.
+    ///
+    pub def range(b: Int32, e: Int32): Iterator[Int32] & Impure = ???
+
+    ///
+    /// Returns an iterator over an iterable with the element `a` repeated `n` times.
+    ///
+    /// Returns `Done` if `n < 0`.
+    ///
+    pub def repeat(a: a, n: Int32): Iterator[a] & Impure = ???
+
+    ///
+    /// Returns `true` if and only if `iter` contains the element `a`.
+    ///
+    // TODO: Does this consume elements until next() == a?
```

Comment:
Yes, I would say:

Consumes elements of the iterator until an element equal to `a` is found.